### PR TITLE
fix(docs): correct and validate component meta information (closes #2665)

### LINKED ISSUE #2665: Component import paths in docs are sometimes incorrect
On https://bootstrap-vue.js.org/docs/components/card/
```
Component | Import Path
-- | --
<b-card> | bootstrap-vue/es/components/card/card
<b-card-header> | bootstrap-vue/es/components/card-header/card-header
<b-card-footer> | bootstrap-vue/es/components/card-footer/card-footer
<b-card-body> | bootstrap-vue/es/components/card-body/card-body
<b-card-title> | bootstrap-vue/es/components/card-title/card-title
<b-card-sub-title> | bootstrap-vue/es/components/card-sub-title/card-sub-title
<b-card-img> | bootstrap-vue/es/components/card-img/card-img
<b-card-text> | bootstrap-vue/es/components/card-text/card-text
<b-card-group> | bootstrap-vue/es/components/card-group/card-group
```

Import paths are incorrect, i.e. for header it should be `bootstrap-vue/es/components/card/card-header`

### PR DESCRIPTION
**Pull request by pimlie**
## Description of Pull Request:

The `Importing individual <plugin> Components` overview section in the docs is showing the wrong paths as it uses the component name as plugin dir, which is a problem for plugins that provide multiple components. 

Before:
image

After:
image

Closes #2665.

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [X] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact:

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch, **not** the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (i.e.
      `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [ ] The PR should address only one issue or feature. If adding multiple features or fixing a bug
      and adding a new feature, break them into separate PRs if at all possible.
- [X] PR titles should following the
      [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e.
      "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos",
      "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated
      from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and
      event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or
      keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a
      suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by pimlie** [comment]
Just noticed this fix is not enough, this pr will still leave the wrong paths for `image` (it will show `img` instead). 
Please leave this PR open, I will push a script to check all meta info shortly. Already half way done with that.

File: docs/components/importdoc.vue [hunk 1]
```diff
@@ -129,7 +132,7 @@ export default {
     pluginImportCode() {
       const pluginLocation = this.isComponentRoute ? 'components' : 'directives'
       return [
-        `import { ${this.pluginName} } from 'bootstrap-vue/es/${pluginLocation}'`,
+        `import ${this.pluginName} from 'bootstrap-vue/es/${pluginLocation}/${this.pluginDir}'`,
```
Comment (jackmu95) [inline docs/components/importdoc.vue]: Please add the brackets ` { ${this.pluginName} }` back here.

File: scripts/check-plugin-meta.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env node
+
+const fs = require('fs')
```
Comment (jackmu95) [inline scripts/check-plugin-meta.js]: Please use `import` here.

File: scripts/check-plugin-meta.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      if (stats.isDirectory()) {
+        const files = await readDir(pluginDir)
+        const componentModules = []
+        files.forEach(file => {
+          if (file.startsWith(plugin) && !file.includes('.spec.')) {
+            componentModules.push(file.replace(/\.js/, ''))
+          }
+        })
+
+        const pckg = require(`${pluginDir}/package.json`)
```
Comment (jackmu95) [inline scripts/check-plugin-meta.js]: ``const { meta } = require(`${pluginDir}/package.json`)``

File: scripts/check-plugin-meta.js (hunk 2 above)
Comment (pimlie) [reply scripts/check-plugin-meta.js]: If we do that then we need to have `esm` as a dev dependency and you need to run the script with `node -r esm`

Comment (jackmu95) [reply scripts/check-plugin-meta.js]: That's a point.

Comment (pimlie) [reply scripts/check-plugin-meta.js]: Damn, I just committed those changes. Adding esm as dev dependency shouldnt really be an issue as nuxt also uses it so we already have it anyway.

Comment (pimlie) [reply scripts/check-plugin-meta.js]: Also now I can import the startCase util from src and dont have to copy that implementation :)

Comment (jackmu95) [reply scripts/check-plugin-meta.js]: Ok. I'am fine with that :)

File: docs/components/importdoc.vue (hunk 1 above)
Comment (pimlie) [reply docs/components/importdoc.vue]: Are you sure about this? It was actually Troy who removed this in 8af2743f90abc189a4269b1bba4a471020fbdafa as he added the `${this.pluginDir}` which has a default install export. Probably after my question yesterday as this syntax works better with tree shaking.

**Comment by pimlie** [comment]
Btw, my remark about the invalid path from `image` is invalid. Probably had too many tabs open and was looking at the live docs instead of dev docs. Sorry about that.

**Comment by tmorehouse** [comment]
For component `package.json` files, I was contemplating deprecation of the `component` key in `meta` in favour of using `components` key as an array of components (even if only a single component), just to make the logic simpler in the `componentDoc.vue` doc generator.

**Comment by jackmu95** [comment]
@tmorehouse We can do that for sure 👍

**Comment by jackmu95** [comment]
@tmorehouse I've noticed that there are some event or slots missing for some components but this PR is huge enough already. We should do that later on.

File: src/components/modal/package.json [hunk 4]
```diff
@@ -187,6 +187,10 @@
           {
             "name": "modal-cancel",
             "description": "Modal CANCEL button content."
+          },
+          {
+              "name": "modal-backdrop",
+              "description": "Modal Backdrop content."
```
Comment (tmorehouse) [inline src/components/modal/package.json]: May want to remove the extra spaces in front of the keys (2 spaces)

**Comment by jackmu95** [comment]
Thanks @pimlie!

**Pull request merged**